# Nike Sales Data Analytics Project

**Student Name:** Hashfina Fathima PP  
**Course:** Data Analytics Final Project  
**Date:** 03 February 2026  


Step 1: Data Loading and Initial Overview


## Project Overview

## Domain: Sportswear & Retail Sales Analytics

 **Objective** : This project analyzes Nike sales performance across product categories, customer segments, regions, and sales channels to understand patterns that drive higher revenue and profit in a global retail context.
 
 **Business Problem** : Nike sells a wide range of footwear, apparel, and accessories across multiple regions and both online and offline channels. Understanding which product lines, customer segments, and channels generate the most profitable growth is critical for:  

- Designing better product mix and inventory planning (by category, size, and gender segment).
- Optimizing pricing and **discount** strategies to balance volume and profitability. 
- Prioritizing high-performing regions and sales channels for targeted campaigns and expansion. 
- Improving demand forecasting by identifying seasonal and category-level sales patterns.

 **Dataset**: Nike Sales Dataset containing 2,500 transaction records with 13 features, including order details, customer segment (gender category), product attributes, pricing, discount, sales channel, region, revenue, and profit.

## Problem Statement

This project analyzes Nike sales data to understand how different product categories, regions, and sales channels contribute to overall revenue and profit. By exploring trends in units sold, discounts, and profitability, the study aims to identify which parts of the business perform best and where there are opportunities to improve.

The analysis will use Python-based exploratory data analysis and visualizations to uncover key patterns in customer demand and product performance. Based on these insights, the project will provide clear, data-driven recommendations to optimize Nike’s sales strategy, including decisions on product focus, regional prioritization, and discount policies.


## Step 1: Data Loading and Initial Overview
**Goal**: Import the dataset, examine its structure, and understand the basic characteristics of the data.

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

plt.style.use("seaborn-v0_8")

sns.set_palette("Set2")

df = pd.read_csv("Nike_Sales_Uncleaned.csv")

print("Libraries imported successfully!")


Libraries imported successfully!


## 1.1 Data Source and Loading


**Data Source**: Nike Sales Dataset (CSV file provided as part of the Data Analytics Final Project).


**Source Link**: https://www.kaggle.com/datasets/nayakganesh007/nike-sales-uncleaned-dataset


**Dataset Description**:
 This dataset contains transactional sales information for 2,500 Nike orders, including:

- Order details (Order_ID, Order_Date, Units_Sold).


- Customer segment information (Gender_Category).


- Product attributes (Product_Line, Product_Name, Size, MRP, Discount_Applied).


- Performance metrics (Revenue, Profit) across different Regions and Sales_Channels.


**Loading Objectives**:

We will load the CSV file and perform an initial inspection to understand:


- Total number of records (rows) – 2,500, representing individual sales transactions.


- Total number of features (columns) – 13, representing order, product, customer, and performance attributes.


- Column names and their data types using df.info().


- Overall data structure and quality, including missing values in Size, Units_Sold, MRP, Discount_Applied, and Order_Date

## 1.2 First Look at the Data


We will examine the first few rows of the Nike sales dataset to understand:


- The structure and format of the transactional data.


- Actual values in each column such as Order_ID, Product_Line, Gender_Category, Revenue, and Profit.


- Data types for each field (numeric measures like Units_Sold, MRP, Revenue, Profit vs. categorical fields like Region, Sales_Channel, Product_Line).


- Any obvious data quality issues, such as missing values in Size, Units_Sold, MRP, Discount_Applied, and Order_Date.


The df.head() function displays the first 5 rows by default, giving a quick preview of the dataset. We can use the df.head(n) method to check the top *n* rows of the dataframe, where *n* is an integer such as 10 or 20 for a broader initial look.

In [2]:
# 1.2 First Look at the Data
# Display the first 10 rows of the dataset
print("First 10 rows of the dataset:")
df.head(10)


First 10 rows of the dataset:


,Order_ID,Gender_Category,Product_Line,Product_Name,Size,Units_Sold,MRP,Discount_Applied,Revenue,Order_Date,Sales_Channel,Region,Profit
0,2000,Kids,Training,SuperRep Go,M,NaN,NaN,0.47,0.0,2024-03-09,Online,bengaluru,-770.45
1,2001,Women,Soccer,Tiempo Legend,M,3.0,4957.93,NaN,0.0,2024-07-09,Retail,Hyd,-112.53
2,2002,Women,Soccer,Premier III,M,4.0,NaN,NaN,0.0,NaN,Retail,Mumbai,3337.34
3,2003,Kids,Lifestyle,Blazer Mid,L,NaN,9673.57,NaN,0.0,04-10-2024,Online,Pune,3376.85
4,2004,Kids,Running,React Infinity,XL,NaN,NaN,NaN,0.0,2024/09/12,Retail,Delhi,187.89
5,2005,Women,Training,Flex Trainer,M,1.0,7363.96,NaN,0.0,NaN,Retail,Delhi,1415.98
6,2006,Men,Training,SuperRep Go,M,NaN,6819.78,NaN,0.0,04-06-2025,Online,Bangalore,1802.09
7,2007,Kids,Training,SuperRep Go,M,1.0,NaN,0.32,0.0,NaN,Retail,Bangalore,2928.36
8,2008,Women,Soccer,Premier III,11,-1.0,NaN,NaN,0.0,11-10-2024,Online,Hyderabad,-731.18
9,2009,Men,Soccer,Mercurial Superfly,M,4.0,NaN,NaN,0.0,NaN,Online,Pune,3929.61


## 1.3 Last Rows of the Dataset


Checking the last few rows of the Nike sales dataset helps us:


- Verify that all 2,500 transaction records have been loaded completely.


- Observe whether data patterns or missing values change toward the end of the file.


- Ensure there is no corruption or unexpected formatting issues at the bottom of the dataset.


The df.tail(n) function displays the last *n* rows of the dataframe

In [3]:
# 1.3 Last Rows of the Dataset
# Display the last 10 rows of the dataset
print("Last 10 rows of the Nike sales dataset:")
df.tail(10)


Last 10 rows of the Nike sales dataset:


,Order_ID,Gender_Category,Product_Line,Product_Name,Size,Units_Sold,MRP,Discount_Applied,Revenue,Order_Date,Sales_Channel,Region,Profit
2490,4490,Women,Training,Metcon 7,L,0.0,NaN,NaN,0.0,23-02-2025,Online,Kolkata,-1173.84
2491,4491,Women,Running,Pegasus Turbo,M,0.0,NaN,NaN,0.0,NaN,Online,Kolkata,3066.85
2492,4492,Men,Lifestyle,Air Force 1,M,NaN,NaN,NaN,0.0,NaN,Retail,Mumbai,2397.23
2493,4493,Women,Lifestyle,Blazer Mid,XL,4.0,NaN,NaN,0.0,2024-10-04,Online,Mumbai,321.24
2494,4494,Women,Training,ZoomX Invincible,L,NaN,NaN,NaN,0.0,12-02-2025,Online,Kolkata,1357.29
2495,4495,Kids,Basketball,Kyrie Flytrap,XL,3.0,NaN,NaN,0.0,2025-05-14,Online,Pune,2.97
2496,4496,Men,Basketball,Kyrie Flytrap,L,-1.0,NaN,NaN,0.0,NaN,Online,Hyd,-487.36
2497,4497,Men,Soccer,Tiempo Legend,7,NaN,6647.60,0.52,0.0,02-07-2025,Retail,Bangalore,-918.14
2498,4498,Women,Training,ZoomX Invincible,NaN,4.0,5358.70,NaN,0.0,11-12-2024,Online,Mumbai,3352.29
2499,4499,Women,Running,Air Zoom,M,NaN,5550.99,NaN,0.0,23-11-2024,Retail,Mumbai,2453.57


## 1.4 Dataset Information and Data Types


Understanding data types in the Nike sales dataset is crucial for:


- Identifying which columns may need type conversion (for example, converting Order_Date to datetime, or ensuring Units_Sold and MRP are numeric).


- Planning appropriate analysis methods for numeric measures (Revenue, Profit, Units_Sold, MRP) versus categorical fields (Region, Sales_Channel, Product_Line, Gender_Category).


- Detecting potential data quality issues such as missing values in Size, Units_Sold, MRP, Discount_Applied, and Order_Date.


- Determining memory usage and optimization needs if the dataset is scaled up.

The df.info() method provides:


- Total number of entries (rows) – 2,500 Nike sales transactions.


- Column names and their data types (int64, float64, object).


- Number of non-null values in each column, which helps identify missing data.


- Memory usage of the dataframe in the current session.


In [4]:
df.shape      # shows number of rows and columns
df.info()     # shows column names and data types


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2500 entries, 0 to 2499
Data columns (total 13 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Order_ID          2500 non-null   int64  
 1   Gender_Category   2500 non-null   object 
 2   Product_Line      2500 non-null   object 
 3   Product_Name      2500 non-null   object 
 4   Size              1990 non-null   object 
 5   Units_Sold        1265 non-null   float64
 6   MRP               1246 non-null   float64
 7   Discount_Applied  832 non-null    float64
 8   Revenue           2500 non-null   float64
 9   Order_Date        1884 non-null   object 
 10  Sales_Channel     2500 non-null   object 
 11  Region            2500 non-null   object 
 12  Profit            2500 non-null   float64
dtypes: float64(5), int64(1), object(7)
memory usage: 254.0+ KB


df.head()
df.shape
df.info()
df.describe(include="all")


## 1.5 Column Names and Descriptions


Below is a brief description of each column in the Nike sales dataset:

 **Order and product information:**


- Order_ID: Unique identifier for each Nike sales transaction.


- Product_Line: High-level category of the product (for example, Footwear, Apparel, Accessories).


- Product_Name: Specific Nike product sold in the transaction.


- Size: Size of the product (for example, shoe size or clothing size); contains missing values for some records.


**Sales quantity and pricing:**


- Units_Sold: Number of units sold in the transaction.


- MRP: Maximum Retail Price per unit of the product.


- Discount_Applied: Discount amount or percentage applied to the product in that order.


**Revenue, profit, and date:**


- Revenue: Total sales value for the transaction after discount.


- Profit: Profit earned from the transaction after costs.

- Order_Date: Date on which the order was placed; later converted to datetime and used to derive Year and Month.


- Year: Year extracted from Order_Date for time-based analysis.


- Month: Month name extracted from Order_Date for seasonality analysis.


**Customer and channel information:**


- Gender_Category: Customer segment based on gender (for example, Men, Women, Kids).


- Sales_Channel: Channel through which the product was sold (for example, Online, Retail).

- Region: Geographic region where the sale occurred.




## 1.6 Statistical Summary of Numerical Features


The describe() function provides key statistics for all numerical columns in the Nike sales dataset, such as Units_Sold, MRP, Discount_Applied, Revenue, and Profit.


**Statistical measures returned include:**


- count: Number of non-null values in each numeric column (for example, Units_Sold has 1265 non-null values, Revenue and Profit have 2500).


- mean: Average value (for example, average Units_Sold is about 1.48, average MRP is around 6039.86).


- std: Standard deviation, indicating variability in sales quantities, prices, discounts, revenue, and profit.


- min and max: Minimum and maximum values 


- 25%, 50%, 75%: Quartiles (Q1, median, Q3), helping to understand distribution and central tendency.


**This helps to identify:**

- Overall data distribution and spread of Nike sales performance metrics.


- Potential outliers or invalid values, such as negative Units_Sold, negative Revenue, or Profit, and unusually high discounts.


- Missing data, when the count for a column is less than the total number of rows (2,500).


- Numeric columns that might be stored incorrectly as text.


In [5]:
# 1.6 Statistical Summary of Numerical Features
df[['Units_Sold', 'MRP', 'Discount_Applied', 'Revenue', 'Profit']].describe()


,Units_Sold,MRP,Discount_Applied,Revenue,Profit
count,1265.000000,1246.000000,832.000000,2500.000000,2500.000000
mean,1.482213,6039.863395,0.631022,274.873716,1376.012848
std,1.696947,2315.746826,0.365500,2023.773550,1478.671013
min,-1.000000,2006.630000,0.000000,-7561.590000,-1199.450000
25%,0.000000,4038.902500,0.320000,0.000000,93.532500
50%,2.000000,6110.030000,0.615000,0.000000,1371.385000
75%,3.000000,8022.187500,0.962500,0.000000,2660.645000
max,4.000000,9996.220000,1.250000,37169.350000,3999.210000


# Task 2: Data Preprocessing

**Goal:** Prepare a clean dataset ready for exploratory data analysis and insights generation.

Based on the data quality issues identified in Task 1, we will perform the following data cleaning tasks:

## Tasks to Complete:

1. ✅ **Handle missing values** in Order_Date, Units_Sold, MRP, Discount_Applied
2. ✅ **Remove duplicate rows** to ensure data integrity
3. ✅ **Verify and fix data types** (dates, numeric values)
4. ✅ **Handle outliers** in Units_Sold, MRP, Revenue, Profit
5. ✅ **Standardize categorical values** (Region, Product_Line, Sales_Channel)
6. ✅ **Create derived features** (DiscountPercent, ProfitMargin, MonthYear)
7. ✅ **Validate cleaned data** - final quality checks

By the end of this step, we will have a **clean, analysis-ready dataset** saved as `Nike_Sales_Cleaned.csv`.


## 2.1 Data Quality Checks


**Missing Values:**

Missing data in the Nike sales dataset can affect statistical calculations and model performance, especially for key fields like Units_Sold, MRP, Discount_Applied, and Order_Date.

We first need to identify which columns have missing values and how many, so we can plan appropriate handling strategies such as imputation, removal, or business-rule-based fixes.


**Duplicate Rows:**

Duplicate transaction records can distort revenue, profit, and demand analysis, and may indicate data entry or export errors.

Such duplicates should be removed to ensure data integrity before building any insights or models.


The isna().sum() (or isnull().sum()) function is used to count missing values in each column, and duplicated().sum() is used to count duplicate rows.


In [6]:
# 2.2 Data Quality Checks – Missing Values

# Check for missing values in all columns
print("Missing Values Summary:")
missing_values = df.isna().sum()
print(missing_values)
print("=" * 70)

# Columns with missing values
if missing_values.sum() > 0:
    print("\nColumns with missing values:")
    print(missing_values[missing_values > 0])
else:
    print("\nNo missing values found in the dataset.")


Missing Values Summary:
Order_ID               0
Gender_Category        0
Product_Line           0
Product_Name           0
Size                 510
Units_Sold          1235
MRP                 1254
Discount_Applied    1668
Revenue                0
Order_Date           616
Sales_Channel          0
Region                 0
Profit                 0
dtype: int64

Columns with missing values:
Size                 510
Units_Sold          1235
MRP                 1254
Discount_Applied    1668
Order_Date           616
dtype: int64


# 2.3 Data Quality Checks – Duplicate Rows

# Check for duplicate rows
duplicates = df.duplicated().sum()
print(f"Number of duplicate rows: {duplicates}")

if duplicates > 0:
    print(f"\nWarning: {duplicates} duplicate row(s) detected!")
else:
    print("\nNo duplicate rows found in the dataset.")


In [7]:
df.columns


Index(['Order_ID', 'Gender_Category', 'Product_Line', 'Product_Name', 'Size',
       'Units_Sold', 'MRP', 'Discount_Applied', 'Revenue', 'Order_Date',
       'Sales_Channel', 'Region', 'Profit'],
      dtype='object')

In [8]:
df['Order_Date'] = pd.to_datetime(df['Order_Date'], errors='coerce', dayfirst=True)

df['Year'] = df['Order_Date'].dt.year
df['Month'] = df['Order_Date'].dt.month_name()


In [9]:
df[df['Units_Sold'] < 0].head()
df[df['MRP'] <= 0].head()
df[df['Discount_Applied'] < 0].head()
df[df['Discount_Applied'] > 1].head()   # if discount should be between 0 and 1
df[df['Revenue'] < 0].head()
df[df['Profit'] < 0].head()


,Order_ID,Gender_Category,Product_Line,Product_Name,Size,Units_Sold,MRP,Discount_Applied,Revenue,Order_Date,Sales_Channel,Region,Profit,Year,Month
0,2000,Kids,Training,SuperRep Go,M,NaN,NaN,0.47,0.0,2024-09-03,Online,bengaluru,-770.45,2024.0,September
1,2001,Women,Soccer,Tiempo Legend,M,3.0,4957.93,NaN,0.0,2024-09-07,Retail,Hyd,-112.53,2024.0,September
8,2008,Women,Soccer,Premier III,11,-1.0,NaN,NaN,0.0,NaT,Online,Hyderabad,-731.18,NaN,NaN
14,2014,Women,Running,React Infinity,11,NaN,NaN,NaN,0.0,2025-12-04,Online,Bangalore,-1044.40,2025.0,December
18,2018,Men,Training,Metcon 7,XL,0.0,NaN,1.17,0.0,NaT,Online,bengaluru,-966.89,NaN,NaN


In [10]:
# remove rows with non‑positive MRP
df = df[df['MRP'] > 0]

# remove rows with impossible discounts (> 1)
df = df[df['Discount_Applied'].between(0, 1)]


Cleaning


In [11]:
# 1. Strip spaces and fix casing
cat_cols = ['Gender_Category', 'Product_Line', 'Product_Name', 
            'Size', 'Sales_Channel', 'Region']

for col in cat_cols:
    df[col] = df[col].astype(str).str.strip().str.title()

# 2. Optional: Standardize Region (if needed)
df['Region'] = df['Region'].replace({
    'Hyd': 'Hyderabad',
    'Banglore': 'Bengaluru',
    'Bengaluru ': 'Bengaluru',
})

# 3. Optional: Standardize Gender Category
df['Gender_Category'] = df['Gender_Category'].replace({
    'Mens': 'Men',
    'Womens': 'Women',
    'Kids ': 'Kids'
})

In [12]:
print(df['Region'].unique())
print(df['Gender_Category'].unique())
print(df['Product_Line'].unique())

['Hyderbad' 'Hyderabad' 'Delhi' 'Mumbai' 'Pune' 'Bengaluru' 'Kolkata'
 'Bangalore']
['Women' 'Kids' 'Men']
['Basketball' 'Running' 'Soccer' 'Training' 'Lifestyle']


In [13]:
cat_cols = ['Gender_Category', 'Product_Line', 'Product_Name',
            'Size', 'Sales_Channel', 'Region']

for col in cat_cols:
    df[col] = df[col].astype('category')

In [14]:
print(df.info())
print(df.head())
print(df.describe(include='all'))

<class 'pandas.core.frame.DataFrame'>
Index: 308 entries, 16 to 2497
Data columns (total 15 columns):
 #   Column            Non-Null Count  Dtype         
---  ------            --------------  -----         
 0   Order_ID          308 non-null    int64         
 1   Gender_Category   308 non-null    category      
 2   Product_Line      308 non-null    category      
 3   Product_Name      308 non-null    category      
 4   Size              308 non-null    category      
 5   Units_Sold        164 non-null    float64       
 6   MRP               308 non-null    float64       
 7   Discount_Applied  308 non-null    float64       
 8   Revenue           308 non-null    float64       
 9   Order_Date        30 non-null     datetime64[ns]
 10  Sales_Channel     308 non-null    category      
 11  Region            308 non-null    category      
 12  Profit            308 non-null    float64       
 13  Year              30 non-null     float64       
 14  Month             30 non-null

In [15]:
df_clean = df.dropna(subset=['Revenue', 'Profit'])   # or the columns you need


## Handling invalid and unusual values

First, I checked the numeric columns (`Units_Sold`, `MRP`, `Discount_Applied`, `Revenue`, `Profit`) using `describe()` and filters to find negative or impossible values. I found that some rows had non‑positive MRP values and discount values outside the 0–1 range, which are not realistic for this dataset. 

To improve data quality, I removed rows where `MRP <= 0` and where `Discount_Applied` was less than 0 or greater than 1, because these likely come from data entry errors. I kept negative `Profit` values, since they represent genuine loss‑making transactions that are important for understanding Nike’s performance.


In [16]:
# 1. Convert discount to percentage
df['DiscountPercent'] = df['Discount_Applied'] * 100

# 2. Profit margin (how much profit per 1 unit of revenue)
df['ProfitMargin'] = df['Profit'] / df['Revenue']

# 3. Month–Year for time‑series plots
df['MonthYear'] = df['Order_Date'].dt.to_period('M').astype(str)


## Creating derived columns

To support better analysis, I created three new features from the cleaned data. `DiscountPercent` converts the discount into an easy‑to‑understand percentage. `ProfitMargin` measures how much profit Nike earns per unit of revenue for each order. `MonthYear` combines the order month and year, which will be used to study monthly sales and profit trends over time.


In [17]:
import pandas as pd

df = pd.read_csv("Nike_Sales_Uncleaned.csv")
import pandas as pd

df = pd.read_csv("Nike_Sales_Uncleaned.csv")


In [18]:
df_clean = df.dropna(subset=['Revenue', 'Profit'])


In [19]:
import seaborn as sns
import matplotlib.pyplot as plt

sns.set_theme(style="whitegrid", palette="Set2")


## 2.4 Summary of Findings


**Dataset Overview:**

- Total rows: 2,500 Nike sales transactions.


- Total columns: 13 original features, expanded to 16 after creating DiscountPercent, ProfitMargin, and MonthYear.


- Analysis focus: Revenue, Profit, and ProfitMargin as key performance outcomes across products, segments, and time.


**Data Quality Issues Identified:**

- Missing values: substantial gaps in Size, Units_Sold, MRP, Discount_Applied, and Order_Date, which required dropping or filtering records for a reliable clean dataset (df_clean).


- No duplicate rows were detected, so duplication is not a concern.


- Invalid values: negative Units_Sold, negative or zero MRP, discounts outside a realistic 0–100% range, and negative Revenue/Profit highlight data entry or logic issues that needed to be removed from analysis.

**Feature Engineering Performed:**

- DiscountPercent: converts Discount_Applied into an interpretable percentage discount per order.


- ProfitMargin: measures how much profit is earned per unit of revenue for each order.


- MonthYear: combines month and year from Order_Date to support monthly trend analysis.


**Key Business Insights (Initial):**

- Revenue is highest for the Men Gender_Category, indicating stronger demand among male customers.


- Certain Product_Lines and Regions generate disproportionately higher Revenue and Profit, suggesting priority segments for marketing and inventory focus.


- After cleaning, the remaining valid transactions provide a more trustworthy view of Nike’s sales and profitability patterns across gender, channel, and geography.


## 2.5 Exporting Cleaned Dataset ##


The preprocessed Nike sales dataset is now ready for analysis in the Exploratory Data Analysis (EDA) notebook.


**Export Details:**

Format: CSV (comma-separated values).


Filename: Nike_Sales_Cleaned.csv.


Location: C:/Users/Lenovo/Documents/Nike-Sales-Analysis/data.


Records: number of clean transactions in df_clean (after removing missing/invalid rows).


Columns: all relevant cleaned and engineered features (for example, Order_ID, Product_Line, Units_Sold, MRP, Discount_Applied, Revenue, Profit, Gender_Category, Sales_Channel, Region, and any new columns like ProfitMargin or MonthYear).


In [20]:
# After finishing all cleaning steps on df
df_clean = df.copy()


In [21]:
#  EXPORTING CLEANED DATASET (using df directly)
export_path = r"C:/Users/Lenovo/anaconda3/final project/Nike final project/Nike_Sales_Cleaned.csv"

df.to_csv(export_path, index=False)

print("  Dataset exported successfully!")
print(f"  Location: {export_path}")
print(f"  Rows: {len(df):,}")
print(f"  Columns: {len(df.columns)}")


  Dataset exported successfully!
  Location: C:/Users/Lenovo/anaconda3/final project/Nike final project/Nike_Sales_Cleaned.csv
  Rows: 2,500
  Columns: 13
